File: exercism/c/armstrong-numbers/src/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <stdio.h>
#include <math.h>

bool is_armstrong_number(int candidate) {
  if (candidate == 0)
    return true;

  int num_digits = 0;
  while (candidate >= pow(10, num_digits))
    num_digits++;

  int armstrong_sum = 0;
  int base = candidate;
  while (base > 0) {
    int digit = base % 10;
    armstrong_sum += pow(digit, num_digits);
    base /= 10;
  }

  return candidate == armstrong_sum;
}

File: exercism/go/armstrong-numbers/armstrong_numbers.go
package armstrong

import (
	"math"
	"strconv"
)

// IsNumber determines whether a number is an Armstrong Number
func IsNumber(num int) bool {
	if num == 0 {
		return true
	}

	var armstrongSum int
	numDigits := len(strconv.Itoa(num))

	for base := num; base > 0; base /= 10 {
		digit := base % 10
		armstrongSum += int(math.Pow(float64(digit), float64(numDigits)))
	}

	return num == armstrongSum
}
